Makefile

```diff
@@ -0,0 +1,96 @@
+version=3.81
+
+BACKUP_TIME := `date -u +%F-%H%M%S`
+PROJECT_DIR ?= .
+DEPLOY_LOCAL := ${PROJECT_DIR}/deploy/local
+DOCKER_IMAGE_TAG ?= dev
+DOCKER_IMAGE_ROOT := docker.uncharted.software/terarium
+
+.SECONDEXPANSION:
+
+.PHONY: all
+all:
+	@echo "make <cmd>"
+	@echo ""
+	@echo "Commands:"
+	@echo "  clean                    - remove bin, dist and node_modules directories"
+	@echo "  images                   - build docker images of all targets"
+	@echo "  image-<target>           - build docker image of a specific target - see below for list of targets"
+	@echo "  publish                  - build and push dock images for deployment"
+	@echo "  publish-<target>         - build and push dock image for a specific target - see below for list of targets"
+	@echo "  lint                     - lint the source code"
+	@echo "  test                     - test the source code"
+	@echo "Targets building (but not pushing) docker images of separately:"
+	@echo "  $(TARGETS)"
+	@echo ""
+	@echo "Variables:"
+	@echo "  DOCKER_IMAGE_TAG=<tag>         - tag to append to each pushed docker image (default: dev)"
+
+
+
+## Targets
+TARGETS :=
+
+TARGETS += hmi-server
+clean-hmi-server:
+	./gradlew :packages:services:hmi-server:clean
+
+image-hmi-server: clean-hmi-server
+	./gradlew :packages:services:hmi-server:build -Dquarkus.container-image.build=true -Dquarkus.container-image.tag=$(DOCKER_IMAGE_TAG)
+
+publish-hmi-server: clean-hmi-server
+	./gradlew :packages:services:hmi-server:build -Dquarkus.container-image.push=true -Dquarkus.container-image.tag=$(DOCKER_IMAGE_TAG)
+
+publish-native-hmi-server: clean-hmi-server
```

Comment:
What the `native` does?

Reply:
Native Image is a technology to ahead-of-time compile Java code to a standalone executable, called a native image. This executable includes the application classes, classes from its dependencies, runtime library classes, and statically linked native code from JDK. It does not run on the Java VM, but includes necessary components like memory management, thread scheduling, and so on from a different runtime system, called “Substrate VM”. Substrate VM is the name for the runtime components (like the deoptimizer, garbage collector, thread scheduling etc.). The resulting program has faster startup time and lower runtime memory overhead compared to a JVM.

Reply:
Nice like [Deno _Compiling Executables_](https://deno.land/manual@v1.27.0/tools/compiler); Love it!